server/src/main/java/io/deephaven/server/plugin/type/JsTypeDistribution.java

```diff
@@ -0,0 +1,107 @@
+/**
+ * Copyright (c) 2016-2022 Deephaven Data Labs and Patent Pending
+ */
+package io.deephaven.server.plugin.type;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.deephaven.plugin.type.JsTypeBase;
+
+import java.io.BufferedInputStream;
+import java.io.IOException;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.Objects;
+
+/**
+ * A {@link io.deephaven.plugin.type.JsType} implementation sourced from a distribution directory.
+ */
+public final class JsTypeDistribution extends JsTypeBase {
+
+    /**
+     * Creates a new js type distribution from a package.json file. Assumes the parent of {@code packageJson} is the
+     * distribution directory.
+     *
+     * @param packageJson the path
+     * @return the js type distribution
+     * @throws IOException if an I/O exception occurs
+     */
+    public static JsTypeDistribution fromPackageJson(Path packageJson) throws IOException {
```

Comment:
Instead of requiring the path to the `package.json`, should just be the path to the directory containing the `package.json`.

Every package requires a `package.json`, so requiring to specify it in the path is just redundant; likewise we should throw if a directory is passed that does not contain a `package.json`.

Reply:
Ok, that's an easy change to make.